File: src/main/kotlin/red/man10/bungee/manager/command/privateMessage/ReplyCommand.java
package red.man10.bungee.manager.command.privateMessage;

import net.md_5.bungee.api.ChatColor;
import net.md_5.bungee.api.CommandSender;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import red.man10.bungee.manager.Man10BungeePlugin;

public class ReplyCommand extends TellCommand {

    private Man10BungeePlugin plugin;

    public ReplyCommand(Man10BungeePlugin plugin, String name) {
        super(plugin, name);
        this.plugin = plugin;
    }

    @Override
    public void execute(CommandSender sender, String[] args) {

        String recieverName = getHistory(sender.getName());

        // 引数が無いときは、現在の会話相手を表示して終了する。
        if ( args.length == 0 ) {
            if ( recieverName != null ) {
                sendMessage(sender, "§dCurrent Conversation Partner： " + recieverName);
            } else {
                sendMessage(sender, "§c現在の会話相手はいません。\n" +
                        "§cThere is no current conversation partner.");
            }
            return;
        }

        // 送信先プレイヤーの取得。取得できないならエラーを表示して終了する。
        if ( recieverName == null ) {
            sendMessage(sender, ChatColor.RED +
                    "メッセージ送信先が見つかりません。");
            return;
        }
        ProxiedPlayer reciever = plugin.getProxy().getPlayer(
                getHistory(sender.getName()));
        if ( reciever == null ) {
            sendMessage(sender, "§cメッセージ送信先が見つかりません。\n" +
                    "§cThe destination for the message was not found.");
            return;
        }

        // 送信メッセージの作成
        StringBuilder str = new StringBuilder();
        for (int i = 0; i < args.length; i++) {
            str.append(args[i] + " ");
        }
        String message = str.toString().trim();

        // 送信
        sendPrivateMessage(sender, reciever, message);
    }
}

File: src/main/kotlin/red/man10/bungee/manager/command/ReplyCommand.kt
package red.man10.bungee.manager.command

import net.md_5.bungee.api.CommandSender
import red.man10.bungee.manager.Man10BungeePlugin
import red.man10.bungee.manager.Man10BungeePlugin.Companion.sendMessage


class ReplyCommand(override val plugin: Man10BungeePlugin, name: String) : TellCommand(plugin, name) {

    override fun execute(sender: CommandSender?, args: Array<out String>?) {
        val receiverName = getHistory(sender!!.name)

        // 引数が無いときは、現在の会話相手を表示して終了する。
        if (args?.size == 0) {
            if (receiverName != null) {
                sendMessage(sender, "§d現在の会話相手:${receiverName} §dCurrent Conversation Partner： $receiverName");
            } else {
                sendMessage(
                    sender, "§c現在の会話相手はいません。 §cThere is no current conversation partner.")
            }
            return
        }
        // 送信先プレイヤーの取得。取得できないならエラーを表示して終了する。
        if (receiverName == null) {
            sendMessage(
                sender, "§cメッセージ送信先が見つかりません。")
            return
        }
        val receiver = plugin.proxy.getPlayer(getHistory(sender.name))
        if (receiver == null) {
            sendMessage(sender, "§cメッセージ送信先が見つかりません。§cThe destination for the message was not found.")
            return
        }

        // 送信メッセージの作成
        val str = StringBuilder()
        for (i in 0 until args?.size!!) {
            str.append(args[i] + " ")
        }
        val message = str.toString().trim { it <= ' ' }

        // 送信

        // 送信
        sendPrivateMessage(sender, receiver, message)
    }
}